keras/layers/core.py

```diff
@@ -1121,16 +1121,18 @@ def output_shape(self):
 
     def get_output(self, train=False):
         X = self.get_input(train)
-
-        def step(x, states):
-            output = K.dot(x, self.W) + self.b
-            return output, []
-
-        last_output, outputs, states = K.rnn(step, X,
-                                             initial_states=[],
-                                             mask=None)
-        outputs = self.activation(outputs)
-        return outputs
+        x = K.reshape(X, (-1, self.input_shape[-1]))
+        Y = K.dot(x, self.W) + self.b
+        input_length = self.input_shape[1]
+        if input_length:
+            Y = K.reshape(Y, (-1, input_length, self.output_shape[-1]))
+        else:
+            z = X[:, :, 0]
+            z = K.pack([z] * self.output_shape[-1])
+            z = K.permute_dimensions(z, (1, 2, 0))
+            Y = K.reshape(Y, K.shape(z))
```

Comment:
This part of the code isn't very readable and so would benefit from explanatory comments (similar to how there was one comment per line in `get_initial_states` from `Recurrent`).